js/src/Ice/ConnectionI.js

```diff
@@ -65,14 +65,13 @@ class MessageInfo {
 }
 
 export class ConnectionI {
-    constructor(communicator, instance, transceiver, endpoint, incoming, adapter, removeFromFactory, options) {
+    constructor(communicator, instance, transceiver, endpoint, removeFromFactory, options) {
         this._communicator = communicator;
         this._instance = instance;
         this._desc = transceiver.toString();
         this._type = transceiver.type();
         this._endpoint = endpoint;
-        this._incoming = incoming;
-        this._adapter = adapter;
+        this._adapter = null;
         this._removeFromFactory = removeFromFactory;
```

Comment:
We only support outgoing connections in JavaScript. The adapter field is always null on construction for Outgoing connections.

Reply:
There is no null default in JS?

Reply:
The default is **undefined**. The code uses [strict equality](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Strict_equality), which compares **null** and **undefined** as different values.